app/models/coupon.rb

```diff
@@ -7,7 +7,7 @@ class Coupon < ApplicationRecord
 
   has_many :applied_coupons
   has_many :customers, through: :applied_coupons
-  has_many :coupon_plans
+  has_many :coupon_plans, dependent: :destroy
```

Comment:
Not directly related but to keep in mind for the delete plan side: we will have to soft delete these when soft deleting a plan